File: src/main/java/com/cag/cagbackendapi/entities/SkillMemberEntity.java
package com.cag.cagbackendapi.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.Type;

import javax.persistence.*;
import java.util.UUID;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "skill_member")
public class SkillMemberEntity {
    @Id
    @Type(type = "pg-uuid")
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(
            name = "UUID",
            strategy = "org.hibernate.id.UUIDGenerator"
    )
    private UUID skill_member_id;

    @ManyToOne
    @JoinColumn(foreignKey = @ForeignKey(name = "fk_profile"))
    private ProfileEntity profileEntity;

    @ManyToOne
    @JoinColumn(foreignKey = @ForeignKey(name = "fk_skill"))
    private SkillEntity skillEntity;
}

File: src/main/java/com/cag/cagbackendapi/entities/SkillMemberEntity.kt
package com.cag.cagbackendapi.entities

import org.hibernate.annotations.GenericGenerator
import org.hibernate.annotations.Type
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "skill_member")
data class SkillMemberEntity(
        @Id
        @Type(type = "pg-uuid")
        @GeneratedValue(generator = "UUID")
        @GenericGenerator(
                name = "UUID",
                strategy = "org.hibernate.id.UUIDGenerator"
        )
        var skill_member_id: UUID?,

        @ManyToOne
        @JoinColumn(name = "profile_id", referencedColumnName = "profile_id")
        var profileEntity: ProfileEntity?,

        @ManyToOne
        @JoinColumn(name = "skill_id", referencedColumnName = "skill_id")
        var skillEntity: SkillEntity?
)